telethon/helpers.py

```diff
@@ -95,17 +95,23 @@ def strip_text(text, entities):
     return text
 
 
-def retry_range(retries):
+def retry_range(retries, force_retry=True):
     """
     Generates an integer sequence starting from 1. If `retries` is
     not a zero or a positive integer value, the sequence will be
     infinite, otherwise it will end at `retries + 1`.
     """
-    yield 1
+
+    # We need at least one iteration even if the retries are 0
+    # when force_retry is True.
+    if force_retry:
```

Comment:
I just realized [the documentation states](https://docs.telethon.dev/en/latest/modules/client.html#telethon.client.telegrambaseclient.TelegramBaseClient):

> May be set to a negative or None value for infinite retries

Now using -1 will become 0 and perform 0 iterations instead of infinite. Worse, `None` will crash. We need to modify this to account for that.

```suggestion
    if force_retry and not (retries is None or retries < 0):
```

Reply:
Got it.